etheno/jsonrpc.py

```diff
@@ -4,6 +4,59 @@
 from .etheno import EthenoPlugin
 from .utils import format_hex_address
 
+
+# source: https://ethereum.stackexchange.com/a/83855
+import rlp
+from eth_typing import HexStr
+from eth_utils import keccak, to_bytes
+from rlp.sedes import Binary, big_endian_int, binary
+from web3 import Web3
+from web3.auto import w3
+
+
+class Transaction(rlp.Serializable):
+    fields = [
+        ("nonce", big_endian_int),
+        ("gas_price", big_endian_int),
+        ("gas", big_endian_int),
+        ("to", Binary.fixed_length(20, allow_empty=True)),
+        ("value", big_endian_int),
+        ("data", binary),
+        ("v", big_endian_int),
+        ("r", big_endian_int),
+        ("s", big_endian_int),
+    ]
+
+
+def hex_to_bytes(data: str) -> bytes:
+    return to_bytes(hexstr=HexStr(data))
+
+
+def decode_raw_tx(raw_tx: str):
+    tx = rlp.decode(hex_to_bytes(raw_tx), Transaction)
+    hash_tx = Web3.toHex(keccak(hex_to_bytes(raw_tx)))
```

Comment:
`hex_to_bytes` called twice in two consecutive lines.
It might make sense to save the result of a single call in a variable and reuse it.

Reply:
thanks, fixed it